scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala

```diff
@@ -46,13 +49,43 @@ object TinyJoinAndMergeJob {
 class TinyJoinAndMergeJob(args: Args) extends Job(args) {
   import TinyJoinAndMergeJob._
 
-  val people = peopleInput.read.mapTo(0 -> 'id) { v: Int => v }
+  val input1 = joinInput1.read.mapTo(0 -> 'id) { v: Int => v }
+
+  val joinedData = joinInput2.read
+    .mapTo(0 -> 'id) { v: Int => v }
+    .joinWithTiny('id -> 'id, input1)
+
+  val mergerData = mergerInput.read.mapTo(0 -> 'id) { v: Int => v }
+
+  (mergerData ++ joinedData).groupBy('id) { _.size('count) }.write(output)
+}
+
+class TinyJoinAndMergeUnsupportedJob(args: Args) extends Job(args) {
+  import TinyJoinAndMergeJob._
+
+  val input1 = joinInput1.read.mapTo(0 -> 'id) { v: Int => v }
+
+  val joined = joinInput2.read
+    .mapTo(0 -> 'id) { v: Int => v }
+    .joinWithTiny('id -> 'id, input1)
+
+  // merging the output of a hashjoin with one of its inputs is
```

Comment:
this is probably going to break some jobs. What is the work around? Would a `.forceToDisk` fix it? Would you need `.forceToDiskExecution` to put it in a new flow? That is a bummer, because it is obviously plannable. /cc @cwensel